pkg/api/index.go

```diff
@@ -125,6 +132,20 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 	return appLinks, nil
 }
 
+func (hs *HTTPServer) getDashboardUIDBySlug(c context.Context, orgID int64, slug string) (string, error) {
```

Comment:
Nit. Maybe this should live in plugin manager (PluginDashboardBySlug) to filter by plugin ID as well?